src/components/date/date.spec.tsx

```diff
@@ -945,32 +1068,34 @@ describe("Date", () => {
 
   describe("disablePortal", () => {
     it("renders DatePicker as a direct children of StyledDateInput by default", () => {
+      const mockEvent = { target: {} } as React.MouseEvent<HTMLElement>;
       wrapper = render();
       act(() => {
-        wrapper.find(InputIconToggle).props().onMouseDown({ target: {} });
+        wrapper?.find(InputIconToggle)?.props()?.onMouseDown?.(mockEvent);
       });
 
-      expect(wrapper.update().find(DayPicker).exists()).toBe(true);
+      expect(wrapper?.update().find(DayPicker).exists()).toBe(true);
     });
   });
 
   describe("datepicker container", () => {
     it("should be the InputPresentationStyle element", () => {
+      const mockEvent = { target: {} } as React.MouseEvent<HTMLElement>;
       wrapper = render();
       act(() => {
-        wrapper.find(InputIconToggle).props().onMouseDown({ target: {} });
+        wrapper?.find(InputIconToggle)?.props()?.onMouseDown?.(mockEvent);
       });
 
-      expect(wrapper.update().find(DayPicker).exists()).toBe(true);
-      expect(wrapper.find(DatePicker).props().inputElement.current).toBe(
+      expect(wrapper?.update().find(DayPicker).exists()).toBe(true);
+      expect(wrapper?.find(DatePicker).props().inputElement.current).toBe(
         wrapper.find(StyledInputPresentation).getDOMNode()
       );
     });
   });
 
   describe("required", () => {
-    let input;
-    let label;
+    let input: ReactWrapper | null;
```

Comment:
```suggestion
    let input: ReactWrapper;
```

**suggestion** (non-blocking): as in a few comments above, this seems unnecessary. Same on the next line of course.

This would also mean the optional chaining on lines 1107/1111 isn't needed